components/mpas-framework/src/framework/mpas_globalSumMod.F

```diff
@@ -0,0 +1,6491 @@
+!!        multi-field sums, only dimensions up to 5 are supported.
+!!
+!!        All sums should give identical answers independent of domain
+!!        decomposition or ordering of elements.
+!!
+!!        The interfaces are overloaded, so to compute the sum of an
+!!        array or scalar, the call should be as follows with mpiComm
+!!        being the MPI communicator defined for the model or sub-model:
+!!
+!!        For sums of scalars across MPI tasks, use:
+!!           sum = globalSum(scalar, mpiComm)
+!!        For sums of arrays over the full range of array indices, use:
+!!           sum = globalSum(array, mpiComm)
+!!        For sums of arrays over a subset of array indices, use:
+!!           sum = globalSum(array, mpiComm, indexRange)
+!!            where indexRange is an integer array
+!!            of size (2*numDimensions) that contain the start, end
+!!            index of each dimension. For example, an MPAS
+!!            array(nz, nCells) might have an indexRange defined as
+!!                indexRange(1) = kmin, indexRange(2) = kmax,
+!!                indexRange(3) = 1,    indexRange(4) = nCellsOwned
+!!
+!!        To compute a sum over products (eg for a dot product or for
+!!        using a multiplicative mask), the interface is the same with 
+!!        an extra argument for the second array/scalar, eg:
+!!           sum = globalSum(scalar1, scalar2, mpiComm)
+!!           sum = globalSum(array1, array2, mpiComm)
+!!           sum = globalSum(array1, array2, mpiComm, indexRange)
+!!
+!!        To compute sums of multiple fields, the interfaces are the
+!!        same as above with an added nFld in the name, so:
+!!           sum(1:nFlds) = globalSumNfld(scalar(:), mpiComm)
+!!           sum(1:nFlds) = globalSumNfld(array, mpiComm)
+!!           sum(1:nFlds) = globalSumNfld(array, mpiComm, indexRange)
+!!           sum(1:nFlds) = globalSumNfld(scalar1(:), scalar2(:), mpiComm)
+!!           sum(1:nFlds) = globalSumNfld(array1, array2, mpiComm)
+!!           sum(1:nFlds) = globalSumNfld(array1, array2, mpiComm, indexRange)
+!!        The multiple fields must be in an array with the field index
+!!        as the last (right-most) index in the array. If an indexRange
+!!        is provided, the same range is used for all fields. For the
+!!        sum of a product - the two product arrays must have the same
+!!        shape and size (including number of fields)
+!!
+!!        For reproducibility, single precision sums are computed in
+!!        double precision. For double precision sums, we use the
+!!        double-double algorithm of Donald Knuth, further improved by
+!!        David A Bailey and presented in:
+!!        He, Yun and Chris Ding, 2001: Using Accurate Arithmetics to
+!!          Improve Numerical Reproducibility and Stability in Parallel
+!!          Applications, J. of Supercomputing, 18, 259-277.
+!
+!***********************************************************************
+
+module mpas_globalSumMod
```

Comment:
All other module names and file names are underscore, not camelcase, and the ending Mod is assumed i.e. `mpas_global_sum`

Reply:
I can change to underscore, but the module name needs to be different from the generic interface name so I appended the mod.